activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/ProcessVariablesHelper.java

```diff
@@ -0,0 +1,129 @@
+    }
+    
+    public String getDateTimeFormat() {
+        return dateTimeFormat;
+    }
+    
+    public void setDateTimeFormat(String dateTimeFormat) {
+        this.dateTimeFormat = dateTimeFormat;
+    }
+    
+    public TimeZone getTimeZone() {
+        return timeZone;
+    }
+   
+    public void setTimeZone(TimeZone timeZone) {
+        this.timeZone = timeZone;
+    }
+    
+    private Optional<Map<String, VariableDefinition>> getVariableDefinitionMap(String processDefinitionKey) {
+        ProcessExtensionModel processExtensionModel = processExtensionModelMap.get(processDefinitionKey);
+
+        return Optional.ofNullable(processExtensionModel)
+                .map(ProcessExtensionModel::getExtensions)
+                .map(Extension::getProperties);
+    }
+    
+    private Date convertObject2Date(Object value) throws Exception{
+        Date d = null;
+        if (value instanceof Date) {
+            return (Date)value;
+        }
+        if (value instanceof String) {
+            SimpleDateFormat format = new SimpleDateFormat(dateTimeFormat);
+            format.setTimeZone(timeZone);
+            d = format.parse(String.valueOf(value)); 
+        }
+        if (value instanceof Long) {
+            d = new Date((long)value);                       
+        }
+        return d;
+    }
+    
+    public List<ActivitiException> checkPayloadVariables(SetProcessVariablesPayload setProcessVariablesPayload,
+                                                         String processDefinitionKey) {
+           
+           final String errorMessage = "Variable with name {0} does not exists.";
+           final Optional<Map<String, VariableDefinition>> variableDefinitionMap = getVariableDefinitionMap(processDefinitionKey);
+           List<ActivitiException> activitiExceptions = new ArrayList<>();
+           
+           if (variableDefinitionMap.isPresent()) {
+               
+               Map<String, Object> variablePayloadMap = setProcessVariablesPayload.getVariables();
+              
+               for (Map.Entry<String, Object> var : variablePayloadMap.entrySet()) {
+                   String name = var.getKey();
+                   Object value = var.getValue();
+                   boolean found = false;
+                   for (Map.Entry<String, VariableDefinition> entry : variableDefinitionMap.get().entrySet()) {
+                       
+                       if (entry.getKey().equals(name)) {
+                           String type = entry.getValue().getType();
+                           
+                           if ("date".equals(type) &&  value != null) {
+                               try {
+                                   Date d = convertObject2Date(value);
+                                   if (d != null) {
+                                       var.setValue(value = d);
+                                   }   
+                               } catch (Exception e) {}
```

Comment:
should we log anything in?

Reply:
or add an activitiexception?

Reply:
I have tried to handle this in a better way